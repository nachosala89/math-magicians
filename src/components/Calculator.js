import React from 'react';
import Button from './Button';

// eslint-disable-next-line react/prefer-stateless-function
export default class Calculator extends React.Component {
  render() {
    return (
      <div className="container">
        <div className="row">
          <span className="result text-end col-md-4 py-3">0</span>
        </div>
        <div className="row">
          <Button btnClass="gray-btn" input="A/C" />
          <Button btnClass="gray-btn" input="+/-" />
          <Button btnClass="gray-btn" input="%" />
          <Button btnClass="orange-btn" input="รท" />
        </div>
        <div className="row">
          <Button btnClass="gray-btn" input={7} />
          <Button btnClass="gray-btn" input={8} />
          <Button btnClass="gray-btn" input={9} />
          <Button btnClass="orange-btn" input="ร" />
        </div>
        <div className="row">
          <Button btnClass="gray-btn" input={4} />
          <Button btnClass="gray-btn" input={5} />
          <Button btnClass="gray-btn" input={6} />
          <Button btnClass="orange-btn" input="-" />
        </div>
        <div className="row">
          <Button btnClass="gray-btn" input={1} />
          <Button btnClass="gray-btn" input={2} />
          <Button btnClass="gray-btn" input={3} />
          <Button btnClass="orange-btn" input="+" />
        </div>
        <div className="row">
          <Button btnClass="gray-btn col-md-2" input={0} />
          <Button btnClass="gray-btn" input="." />
          <Button btnClass="orange-btn" input="=" />
        </div>
      </div>
    );
  }
}
