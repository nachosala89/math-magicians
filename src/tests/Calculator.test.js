import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('renders correctly', () => {
  const calculator = renderer
    .create(<Calculator />)
    .toJSON();
  expect(calculator).toMatchSnapshot();
});