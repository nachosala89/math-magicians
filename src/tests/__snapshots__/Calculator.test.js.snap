// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calc-container"
>
  <div
    className="d-flex result calc-row align-items-center"
  >
    <span
      className="w-100 text-end"
    >
      0
    </span>
  </div>
  <div
    className="d-flex calc-row"
  >
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calc-btn orange-btn w-25"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="d-flex calc-row"
  >
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calc-btn orange-btn w-25"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="d-flex calc-row"
  >
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calc-btn orange-btn w-25"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="d-flex calc-row"
  >
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calc-btn orange-btn w-25"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="d-flex calc-row"
  >
    <button
      className="calc-btn gray-btn w-50"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="calc-btn gray-btn w-25"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calc-btn orange-btn w-25"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
